conn.go

```diff
@@ -347,6 +376,97 @@ func (conn *Conn) inWorker() {
 	}
 }
 
+// sigWorker runs in an own goroutine, getting signals and handling
```

Comment:
This comment is confusing - it's not inherently running in an independent goroutine. Are you trying to say that it *should* run in its own goroutine?